client/src/main/java/io/pravega/client/stream/impl/StreamSegments.java

```diff
@@ -89,6 +100,8 @@ public StreamSegments withReplacementRange(StreamSegmentsWithPredecessors replac
      * @param replacementSegments The StreamSegmentsWithPredecessors to verify
      */
     private void verifyReplacementRange(StreamSegmentsWithPredecessors replacementSegments) {
+        log.debug("Verification of replacement segments {} with the current segments {}",
```

Comment:
Let's add a toString() on replacementSegments so this is a meaningful

Reply:
added it using @Lombok annotation.